app/src/main/java/programmers/java/calulator/console/menu/MenuHandler.java

```diff
@@ -0,0 +1,51 @@
+package programmers.java.calulator.console.menu;
+
+import programmers.java.calulator.common.calculator.Calculator;
+import programmers.java.calulator.common.command.Command;
+import programmers.java.calulator.common.reader.Reader;
+import programmers.java.calulator.common.repository.Repository;
+import programmers.java.calulator.common.writer.Writer;
+import programmers.java.calulator.common.command.factory.CommandType;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class MenuHandler {
+    private final Writer writer;
+    private final Map<String, Command> commands;
+    private final Reader reader;
+    private final Calculator calculator;
+    private final Repository repository;
+
+    public MenuHandler(Writer writer, Calculator calculator, Reader reader, Repository repository) {
+        this.writer = writer;
+        this.commands = new HashMap<>();
+        this.reader = reader;
+        this.calculator = calculator;
+        this.repository = repository;
+        initializeCommands();
+    }
+
+    private void initializeCommands() {
+        Stream.of(CommandType.values()).forEach(commandType ->
+                commands.put(commandType.getCommand(), commandType.createCommand(writer, calculator, reader, repository))
+        );
+    }
```

Comment:
```suggestion
    public MenuHandler(Writer writer, Calculator calculator, Reader reader, Repository repository) {
        this.writer = writer;
        this.commands = initializeCommands();
        this.reader = reader;
        this.calculator = calculator;
        this.repository = repository;
        
    }

    private Map<String, Command> initializeCommands() {
	...
    }
```

생성과 초기화가 분리되어 있는 것 보다는 생성과 함께 초기화 되는 것이 좋아보이네요

Reply:
넵 확실히 아래의 방법이 좋다고 생각합니다. 감사합니다!